tests/container-copy

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+set -eu
+
+# Install and setup LXD.
+install_lxd
+lxd init --auto
+
+cp /root/go/bin/lxd /var/snap/lxd/common/lxd.debug
+systemctl reload snap.lxd.daemon
+lxd waitready --timeout 100
+
+# Create a loop device that we can mount as a source for a dir storage pool.
+test_dir="$(mktemp -d XXX.test)"
+loopimg="$(mktemp "${test_dir}/XXX.img")"
+truncate -s 5G "${loopimg}"
+loopdev="$(losetup --show --find "${loopimg}")"
+mkfs.xfs "${loopdev}"
+disk_dir="$(mktemp -d "${test_dir}/XXX.disk")"
+mount "${loopdev}" "${disk_dir}"
+
+# Begin Test.
+set -x
+
+# Create a storage pool with the mounted disk.
+lxc storage create dir dir source="${PWD}/${disk_dir}"
+
+# Create a container which we will copy to another target.
+lxc launch "${TEST_IMG:-ubuntu-minimal-daily:22.04}" c1 --storage dir
+
+# Launch a VM which will be the target of the copy.
+lxc launch "${TEST_IMG:-ubuntu-minimal-daily:22.04}" target --vm
+waitInstanceReady target
+
+# Wait for snapd in the VM to be ready.
+lxc exec target -- sh -ceu "
+if timeout 60 snap wait system seed.loaded; then
+  exit 0
+fi
+exit 1
+"
+
+# Add the target VM as a remote LXD.
+target_addr="$(lxc ls -f csv -c 4 target | grep -v lxdbr | cut -d' ' -f1 | sed 's/"//g')"
+lxc exec target -- snap install lxd --channel "${LXD_SNAP_CHANNEL}"
+lxc exec target -- lxd init --auto
+lxc exec target -- lxc config set core.https_address=:8443
+lxc exec target -- lxc config set core.trust_password=lxd123
+lxc remote add target "https://${target_addr}:8443" --accept-certificate --password "lxd123"
```

Comment:
I'd rather avoid adding dependency on password auth at this point. It is already waiting in the death row ;)

Reply:
I've updated this to use cert auth. In case the test was run without LXD having generated the config directory, I first have to call `lxc remote add <something>` to get the client to create certs.

Reply:
Thanks, I should have provided you with a pointer to pylxd that does just that (https://github.com/canonical/pylxd/blob/main/integration/run-integration-tests#L18-L19) sorry.

Reply:
> I'd rather avoid adding dependency on password auth at this point. It is already waiting in the death row ;)

@simondeziel we'll need to keep tests for password auth at least for LXD 5.0 and LXD 5.21 btw to avoid regressions as its still supported in those series. Please can you ensure we are still testing it somewhere in the snap based tests. Thanks

Reply:
Maybe here would be a good place to do it.